qa/integration-tests/src/test/java/io/camunda/it/migration/util/ZeebeUtil.java

```diff
@@ -0,0 +1,157 @@
+import io.camunda.zeebe.qa.util.cluster.TestZeebePort;
+import io.zeebe.containers.ZeebeContainer;
+import io.zeebe.containers.ZeebeVolume;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Consumer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+
+public class ZeebeUtil {
+
+  private static final String CAMUNDA_OLD_VERSION = "8.6.6";
+  private final ZeebeVolume volume;
+  private final Network network;
+  private ZeebeContainer zeebeContainer;
+  private ZeebeClient zeebeClient;
+  private TestStandaloneBroker broker;
+  private final Path zeebeDataPath;
+
+  public ZeebeUtil(final Network network) {
+    volume = ZeebeVolume.newVolume();
+    zeebeDataPath = Path.of(System.getProperty("user.dir") + "/zeebe-data" + volume.getName());
+    this.network = network;
+  }
+
+  public String getZeebeGatewayAddress() {
+    if (zeebeContainer != null && zeebeContainer.isStarted()) {
+      return zeebeContainer.getInternalAddress(TestZeebePort.GATEWAY.port());
+    }
+    return "host.testcontainers.internal:" + broker.mappedPort(TestZeebePort.GATEWAY);
+  }
+
+  public String getZeebeRestAddress() {
+    if (zeebeContainer != null && zeebeContainer.isStarted()) {
+      return zeebeContainer.getInternalAddress(TestZeebePort.REST.port());
+    }
+    return "http://host.testcontainers.internal:" + broker.mappedPort(TestZeebePort.REST);
+  }
+
+  public ZeebeContainer start86Broker(final Map<String, String> env) {
+    zeebeContainer =
+        new ZeebeContainer(DockerImageName.parse("camunda/zeebe:" + CAMUNDA_OLD_VERSION))
+            .withExposedPorts(26500, 9600, 8080)
+            .withNetwork(network)
+            .withNetworkAliases("zeebe");
+
+    env.forEach(zeebeContainer::withEnv);
+
+    zeebeContainer
+        .withCreateContainerCmdModifier(
+            cmd -> {
+              cmd.withUser("1001:0").getHostConfig().withBinds(volume.asZeebeBind());
+            })
+        .withCommand(
+            "sh", "-c", "chmod -R 777 /usr/local/zeebe/data && /usr/local/bin/start-zeebe");
+
+    zeebeContainer.start();
+
+    zeebeClient =
+        ZeebeClient.newClientBuilder()
+            .gatewayAddress(zeebeContainer.getExternalGatewayAddress())
+            .restAddress(
+                URI.create(
+                    "http://"
+                        + zeebeContainer.getExternalHost()
+                        + ":"
+                        + zeebeContainer.getMappedPort(8080)))
+            .usePlaintext()
+            .build();
+
+    return zeebeContainer;
+  }
+
+  public ZeebeClient getZeebeClient() {
+    return zeebeClient;
+  }
+
+  private void extractVolume() {
+    try {
+      volume.extract(zeebeDataPath);
+      // Need to remove the .topology.meta to be recreated by new broker
+      Files.delete(zeebeDataPath.resolve("usr/local/zeebe/data/.topology.meta"));
```

Comment:
❓ Out of curiosity what happens if we don't do it?

Reply:
If you don't remove it, the new Broker tries to access the file, but the file permissions set by the previous one do not allow it. Don't know if there is a workaround for this though.